src/mumble/AudioOutput.cpp

```diff
@@ -177,23 +182,49 @@ void AudioOutput::addFrameToBuffer(ClientUser *sender, const Mumble::Protocol::A
 	qrwlOutputs.unlock();
 }
 
-void AudioOutput::removeBuffer(const ClientUser *user) {
-	removeBuffer(qmOutputs.value(user));
-}
-
-void AudioOutput::removeBuffer(AudioOutputUser *aop) {
+void AudioOutput::handleInvalidatedBuffer(AudioOutputUser *buffer) {
 	QWriteLocker locker(&qrwlOutputs);
 	QMultiHash< const ClientUser *, AudioOutputUser * >::iterator i;
 	for (i = qmOutputs.begin(); i != qmOutputs.end(); ++i) {
-		if (i.value() == aop) {
+		if (i.value() == buffer) {
 			qmOutputs.erase(i);
-			delete aop;
+			delete buffer;
 			break;
 		}
 	}
 }
 
-AudioOutputSample *AudioOutput::playSample(const QString &filename, float volume, bool loop) {
+void AudioOutput::handlePositionedBuffer(AudioOutputUser *buffer, float x, float y, float z) {
+	qrwlOutputs.lockForRead();
```

Comment:
You should be using a RAII locker (e.g. lock_guard or in this case the respective Qt analogue)

Reply:
Is QReadLocker fine?